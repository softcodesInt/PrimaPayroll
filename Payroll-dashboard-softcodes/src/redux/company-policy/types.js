/**
 * Actions
 * Object
 */
const CompanyPolicyStructure = {
	ADD_COMPANY_POLICY: 'ADD_COMPANY_POLICY',
	GET_COMPANY_POLICY: 'GET_COMPANY_POLICY',
	EDIT_COMPANY_POLICY: 'EDIT_COMPANY_POLICY',
	DEACTIVATE_COMPANY_POLICY: 'DEACTIVATE_COMPANY_POLICY',
	ADD_HOLIDAYS: 'ADD_HOLIDAYS',
	GET_HOLIDAYS: 'GET_HOLIDAYS',
	EDIT_HOLIDAYS: 'EDIT_HOLIDAYS',
	DEACTIVATE_HOLIDAYS: 'DEACTIVATE_HOLIDAYS',
	ADD_TAX_TABLE: 'ADD_TAX_TABLE',
	GET_TAX_TABLE: 'GET_TAX_TABLE',
	EDIT_TAX_TABLE: 'EDIT_TAX_TABLE',
	DEACTIVATE_TAX_TABLE: 'DEACTIVATE_TAX_TABLE',
	GET_PAY_PERIOD: 'GET_PAY_PERIOD',
	GET_BY_ID: 'GET_BY_ID',
	GET_ERROR: 'GET_ERROR',
};

export default CompanyPolicyStructure;
