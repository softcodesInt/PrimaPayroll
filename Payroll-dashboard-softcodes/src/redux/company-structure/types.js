/**
 * Actions
 * Object
 */
const CompanyStructureActions = {
	ADD_HEAD_OR_ITEM: 'ADD_HEAD_OR_ITEM',
	GET_ALL_COMPANY_STRUCTURE: 'GET_ALL_COMPANY_STRUCTURE',
	GET_HEAD: 'GET_HEAD',
	EDIT_HEAD: 'EDIT_HEAD',
	GET_BY_ID: 'GET_BY_ID',
	GET_ITEM: 'GET_ITEM',
};

export default CompanyStructureActions;
