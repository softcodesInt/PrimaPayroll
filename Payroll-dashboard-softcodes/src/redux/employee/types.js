/**
 * Actions
 * Object
 */
const EmployeeStructure = {
	ADD_EMPLOYEE: 'ADD_EMPLOYEE',
	GET_EMPLOYEES: 'GET_EMPLOYEES',
	GET_EMPLOYEE: 'GET_EMPLOYEE',
	DEACTIVATE_EMPLOYEE: 'DEACTIVATE_EMPLOYEE',
	EDIT_EMPLOYEE: 'EDIT_EMPLOYEE',
	GET_EMPLOYEE_PAYSLIP: 'GET_EMPLOYEE_PAYSLIP',
	DOWNLOAD_PAYSLIP: 'DOWNLOAD_PAYSLIP',
	SEND_INVITATION: 'SEND_INVITATION',
	GET_ERROR: 'GET_ERROR',
};

export default EmployeeStructure;
