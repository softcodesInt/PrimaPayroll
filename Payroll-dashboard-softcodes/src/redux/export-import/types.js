/**
 * Actions
 * Object
 */
const ExportImportStructure = {
	EXPORT: 'EXPORT',
	IMPORT: 'IMPORT',
	DOWNLOAD_BACKUP: 'DOWNLOAD_BACKUP',
	IMPORT_BACKUP: 'IMPORT_BACKUP',
	GET_ERROR: 'GET_ERROR',
};

export default ExportImportStructure;
