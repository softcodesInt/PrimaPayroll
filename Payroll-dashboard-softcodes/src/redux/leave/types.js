/**
 * Actions
 * Object
 */
const LeaveStructure = {
	ADD_LEAVE: 'ADD_LEAVE',
	ADD_CATEGORY: 'ADD_CATEGORY',
	GET_LEAVE: 'GET_LEAVE',
	DEACTIVATE_LEAVE: 'DEACTIVATE_LEAVE',
	DEACTIVATE_CATEGORY: 'DEACTIVATE_CATEGORY',
	EDIT_LEAVE: 'EDIT_LEAVE',
	GET_BY_ID: 'GET_BY_ID',
	GET_CATEGORY: 'GET_CATEGORY',
	EDIT_CATEGORY: 'EDIT_CATEGORY',
	GET_EMPLOYEE_AVAILABLE_LEAVE: 'GET_EMPLOYEE_AVAILABLE_LEAVE',
	SAVE_EMPLOYEE_WORKFLOW_LEAVE: 'SAVE_EMPLOYEE_WORKFLOW_LEAVE',
	GET_EMPLOYEE_LEAVE_REQUEST: 'GET_EMPLOYEE_LEAVE_REQUEST',
	APPROVE_REJECT_LEAVE_REQUEST: 'APPROVE_REJECT_LEAVE_REQUEST',
	GET_ERROR: 'GET_ERROR',
};

export default LeaveStructure;
