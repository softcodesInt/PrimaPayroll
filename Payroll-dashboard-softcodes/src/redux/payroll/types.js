/**
 * Actions
 * Object
 */
const PayrollStructure = {
	ADD_PAYROLL: 'ADD_PAYROLL',
	ADD_CATEGORY: 'ADD_CATEGORY',
	GET_PAYROLL: 'GET_PAYROLL',
	DEACTIVATE_PAYROLL: 'DEACTIVATE_PAYROLL',
	DEACTIVATE_CATEGORY: 'DEACTIVATE_CATEGORY',
	EDIT_PAYROLL: 'EDIT_PAYROLL',
	GET_BY_ID: 'GET_BY_ID',
	GET_CATEGORY: 'GET_CATEGORY',
	EDIT_CATEGORY: 'EDIT_CATEGORY',
	ADD_REMUNERATION: 'ADD_REMUNERATION',
	GET_REMUNERATION: 'GET_REMUNERATION',
	GET_REMUNERATION_BY_ID: 'GET_REMUNERATION_BY_ID',
	EDIT_REMUNERATION: 'EDIT_REMUNERATION',
	GET_PAY_PERIOD: 'GET_PAY_PERIOD',
	ACTIVATE_PAY_PERIOD: 'ACTIVATE_PAY_PERIOD',
	GET_BANK_LETTER: 'GET_BANK_LETTER',
	ADD_TRANSACTION: 'ADD_TRANSACTION',
	GET_TRANSACTION: 'GET_TRANSACTION',
	GET_EMPLOYEE_DRIVEN_TAX: 'GET_EMPLOYEE_DRIVEN_TAX',
	ADD_EMPLOYEE_DRIVEN_TAX: 'ADD_EMPLOYEE_DRIVEN_TAX',
	GET_PAYROLL_READY_FOR_TRANSACTION: 'GET_PAYROLL_READY_FOR_TRANSACTION',
	GET_EMPLOYEE_DRIVEN_PAYROLL: 'GET_EMPLOYEE_DRIVEN_PAYROLL',
	ADD_EMPLOYEE_DRIVEN_PAYROLL: 'ADD_EMPLOYEE_DRIVEN_PAYROLL',
	GET_LOANS: 'GET_LOANS',
	ADD_LOAN: 'ADD_LOAN',
	EDIT_LOAN: 'EDIT_LOAN',
	DEACTIVATE_LOAN: 'DEACTIVATE_LOAN',
	GET_ERROR: 'GET_ERROR',
};

export default PayrollStructure;
