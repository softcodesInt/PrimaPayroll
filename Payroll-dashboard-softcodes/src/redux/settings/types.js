/**
 * Actions
 * Object
 */
const SettingsStructure = {
	ADD_BANK: 'ADD_BANK',
	GET_BANKS: 'GET_BANKS',
	DEACTIVATE_BANK: 'DEACTIVATE_BANK',
	EDIT_BANK: 'EDIT_BANK',
	ADD_JOB_TITLE: 'ADD_JOB_TITLE',
	GET_JOB_TITLES: 'GET_JOB_TITLES',
	DEACTIVATE_JOB_TITLE: 'DEACTIVATE_JOB_TITLE',
	EDIT_JOB_TITLE: 'EDIT_JOB_TITLE',
	ADD_TERMINATE_REASON: 'ADD_TERMINATE_REASON',
	GET_TERMINATE_REASON: 'GET_TERMINATE_REASON',
	DEACTIVATE_TERMINATE_REASON: 'DEACTIVATE_TERMINATE_REASON',
	EDIT_TERMINATE_REASON: 'EDIT_TERMINATE_REASON',
	ADD_REINSTATE_REASON: 'ADD_REINSTATE_REASON',
	GET_REINSTATE_REASON: 'GET_REINSTATE_REASON',
	DEACTIVATE_REINSTATE_REASON: 'DEACTIVATE_REINSTATE_REASON',
	EDIT_REINSTATE_REASON: 'EDIT_REINSTATE_REASON',
	ADD_NATURE_OF_CONTRACT: 'ADD_NATURE_OF_CONTRACT',
	GET_NATURE_OF_CONTRACTS: 'GET_NATURE_OF_CONTRACTS',
	DEACTIVATE_NATURE_OF_CONTRACT: 'DEACTIVATE_NATURE_OF_CONTRACT',
	EDIT_NATURE_OF_CONTRACT: 'EDIT_NATURE_OF_CONTRACT',
	ADD_JOB_GRADE: 'ADD_JOB_GRADE',
	GET_JOB_GRADES: 'GET_JOB_GRADES',
	DEACTIVATE_JOB_GRADES: 'DEACTIVATE_JOB_GRADES',
	EDIT_JOB_GRADES: 'EDIT_JOB_GRADES',
	ADD_PENSION_SETTINGS: 'ADD_PENSION_SETTINGS',
	GET_PENSION_SETTINGS: 'GET_PENSION_SETTINGS',
	GET_ERROR: 'GET_ERROR',
};

export default SettingsStructure;
