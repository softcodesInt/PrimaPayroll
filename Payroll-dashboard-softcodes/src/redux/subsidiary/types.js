/**
 * Actions
 * Object
 */
const SubsidiaryStructure = {
	ADD_SUBSIDIARY: 'ADD_SUBSIDIARY',
	GET_SUBSIDIARY: 'GET_SUBSIDIARY',
	DEACTIVATE_SUBSIDIARY: 'DEACTIVATE_SUBSIDIARY',
	EDIT_SUBSIDIARY: 'EDIT_SUBSIDIARY',
	GET_BY_ID: 'GET_BY_ID',
	GET_ERROR: 'GET_ERROR',
};

export default SubsidiaryStructure;
