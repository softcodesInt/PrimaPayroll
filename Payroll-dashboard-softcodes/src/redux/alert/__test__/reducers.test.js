/* -------------------------- Internal Dependencies ------------------------- */
import alertReducer from '../reducer';
import { SET_ALERT, REMOVE_ALERT } from '../types';

describe('Alert Reducer', () => {
	test('is correct', () => {
		const action = { type: 'dummy_action' };
		const initialState = [];

		expect(alertReducer(undefined, action)).toEqual(initialState);
	});
	test('returns the correct state when alert is success', () => {
		const action = {
			type: SET_ALERT,
			payload: {
				alertType: 'success',
				id: '34sdvsdf2e23',
				msg: 'Successfully rendered actiom',
			},
		};
		const expectedState = [
			{
				alertType: 'success',
				id: '34sdvsdf2e23',
				msg: 'Successfully rendered actiom',
			},
		];

		expect(alertReducer(undefined, action)).toEqual(expectedState);
		expect(alertReducer(undefined, action)).toMatchSnapshot();
	});
	test('returns the correct state when alert is Error', () => {
		const action = {
			type: SET_ALERT,
			payload: {
				alertType: 'error',
				id: '34sdvsdf2e23',
				msg: 'An Error Occured.',
			},
		};
		const expectedState = [
			{
				alertType: 'error',
				id: '34sdvsdf2e23',
				msg: 'An Error Occured.',
			},
		];

		expect(alertReducer(undefined, action)).toEqual(expectedState);
		expect(alertReducer(undefined, action)).toMatchSnapshot();
	});
	test('removes alert', () => {
		const action = {
			type: REMOVE_ALERT,
			payload: '34sdvsdf2e23',
		};
		const expectedState = [];

		expect(alertReducer(undefined, action)).toEqual(expectedState);
		expect(alertReducer(undefined, action)).toMatchSnapshot();
	});
});
