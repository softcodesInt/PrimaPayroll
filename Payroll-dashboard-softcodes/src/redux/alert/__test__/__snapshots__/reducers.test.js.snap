// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Reducer removes alert 1`] = `Array []`;

exports[`Alert Reducer returns the correct state when alert is Error 1`] = `
Array [
  Object {
    "alertType": "error",
    "id": "34sdvsdf2e23",
    "msg": "An Error Occured.",
  },
]
`;

exports[`Alert Reducer returns the correct state when alert is success 1`] = `
Array [
  Object {
    "alertType": "success",
    "id": "34sdvsdf2e23",
    "msg": "Successfully rendered actiom",
  },
]
`;
